### PR TITLE
Latest Updates

### PR DESCRIPTION
**Pull request by awjuliani**
* Project now 2019.4
* ML-Agents now Release 3
* Update project version to 4.0
* Fix typo in name of one of the reset parameters

### CODE REVIEW

File: Assets/ObstacleTower/Scenes/Procedural.unity [hunk 1]
```diff
@@ -689,7 +694,7 @@ MonoBehaviour:
   m_Model: {fileID: 0}
   m_InferenceDevice: 0
   m_BehaviorType: 0
-  m_BehaviorName: ObstacleTowerAgent
+  m_BehaviorName: ObstacleTowerAgent-v4.0
```
Comment (vincentpierre) [inline Assets/ObstacleTower/Scenes/Procedural.unity:697]: Is there a reason for adding the version to the behavior name ?

File: Assets/ObstacleTower/Scripts/AgentLogic/ObstacleTowerManager.cs [hunk 2]
```diff
@@ -77,7 +77,7 @@ private void UpdateEnvironmentParameters()
             Debug.Log("visual-theme outside of valid range. Using default value.");
         }
         
-        var perspective = (int)Academy.Instance.EnvironmentParameters.GetWithDefault("perspective", (int)AgentPerspective.ThirdPerson);
+        var perspective = (int)Academy.Instance.EnvironmentParameters.GetWithDefault("agent-perspective", (int)AgentPerspective.ThirdPerson);
```
Comment (vincentpierre) [inline Assets/ObstacleTower/Scripts/AgentLogic/ObstacleTowerManager.cs:80]: Is there a reason for renaming this environment parameter? Should the documentation change?

Comment (awjuliani) [reply Assets/ObstacleTower/Scripts/AgentLogic/ObstacleTowerManager.cs:80]: This fixes a typo which I think was introduced when you moved the code to the new environment parameter system.

File: Assets/ObstacleTower/Scenes/Procedural.unity (hunk 1 above)
Comment (awjuliani) [reply Assets/ObstacleTower/Scenes/Procedural.unity:697]: We use the version to check for compatibility. In the past this was handled through the academy name, but that was deprecated, so now we check for it here.

File: Assets/ObstacleTower/Scripts/AgentLogic/ObstacleTowerManager.cs (hunk 2 above)
Comment (vincentpierre) [reply Assets/ObstacleTower/Scripts/AgentLogic/ObstacleTowerManager.cs:80]: Arf. Good catch then.

**Pull request merged**